src/vivarium_gates_nutrition_optimization/components/observers.py

```diff
@@ -21,49 +12,62 @@ class ResultsStratifier(ResultsStratifier_):
     def register_stratifications(self, builder: Builder) -> None:
         super().register_stratifications(builder)
 
-        self.setup_stratification(
-            builder,
-            name="pregnancy_outcome",
-            sources=[Source("pregnancy_outcome", SourceType.COLUMN)],
-            categories=models.PREGNANCY_OUTCOMES,
+        builder.results.register_stratification(
+            "pregnancy_outcome",
+            models.PREGNANCY_OUTCOMES,
+            requires_columns=["pregnancy_outcome"],
         )
 
-        self.setup_stratification(
-            builder,
-            name="pregnancy",
-            sources=[Source("pregnancy", SourceType.COLUMN)],
-            categories=models.PREGNANCY_MODEL_STATES,
+        builder.results.register_stratification(
+            "pregnancy", models.PREGNANCY_MODEL_STATES, requires_columns=["pregnancy"]
         )
 
-        self.setup_stratification(
-            builder,
-            name="anemia_status_at_birth",
-            sources=[Source("anemia_status_at_birth", SourceType.COLUMN)],
-            categories=data_values.ANEMIA_STATUS_AT_BIRTH_CATEGORIES,
+        builder.results.register_stratification(
+            "anemia_status_at_birth",
+            data_values.ANEMIA_STATUS_AT_BIRTH_CATEGORIES,
+            requires_columns=["anemia_status_at_birth"],
         )
 
 
 class PregnancyObserver(DiseaseObserver):
     def __init__(self):
         super().__init__("pregnancy")
 
-    def _get_population_view(self, builder: Builder) -> PopulationView:
-        columns_required = [
-            self.current_state_column_name,
-            self.previous_state_column_name,
-            "pregnancy_outcome",
-        ]
-        return builder.population.get_view(columns_required)
-
 
 class MaternalMortalityObserver(MortalityObserver):
-    def on_post_setup(self, event: Event) -> None:
-        self.causes_of_death += ["maternal_disorders"]
+    def setup(self, builder: Builder):
+        super().setup(builder)
+        cause_of_death = models.MATERNAL_DISORDERS_MODEL_NAME
+        self.causes_of_death += cause_of_death
+
+        builder.results.register_observation(
```

Comment:
Is the fact that this required more work than previously the nature of the new system? Or is it more due to you doing custom stuff (I think changing when to register)?

Reply:
the structure of the new results management system exposes the issue here, since more things happen during `setup`, so we have to reproduce them instead of it happening later (which is why we could get away with the change to `on_post_setup`.

that being said, it wouldn't be hard to refactor `self.causes_of_death` into a method that gets called in setup, and then we wouldn't have to copy so much.